### PR TITLE
Writing InstrumentationModule line by line

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2662
And resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/536 -- after this is merged all questions/tasks will have been solved or have issues created for them.

### CODE REVIEW

**Review by trask — APPROVED** [review]
😻

I felt like some sub-section headings would help, it was not always clear when jumping from one topic to another, so I suggested some heading, but not sure if it's too much, very happy with whatever you decide, this doc is 💯

File: docs/contributing/writing-instrumentation-module.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+package conventions (see [muzzle docs](muzzle.md#compile-time-reference-collection) for more info on
+this topic), but it is also possible to explicitly tell which packages/classes are supposed to be
+treated as helper classes by extending one of these methods:
+
+```java
+@Override
+protected String[] additionalHelperClassNames() {
+  return new String[] {"some.additional.HelperClass"};
+}
+
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply docs/contributing/writing-instrumentation-module.md:56]: 💯

**Comment by mateuszrzeszutek** [comment]
> I felt like some sub-section headings would help, it was not always clear when jumping from one topic to another, so I suggested some heading, but not sure if it's too much, very happy with whatever you decide, this doc is 💯

Those headings really make it look better, thanks!

**Comment by trask** [comment]
🥳

**Pull request merged**